clients/admin-ui/src/features/system/AddMultipleSystems.tsx

```diff
@@ -0,0 +1,249 @@
+        innerRowSelection[index] = true;
+      }
+    });
+    return innerRowSelection;
+  }, [dictionaryOptions]);
+
+  const tableInstance = useReactTable<MultipleSystemTable>({
+    columns,
+    data: dictionaryOptions,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    onGlobalFilterChange: setGlobalFilter,
+    enableRowSelection: true,
+    enableSorting: true,
+    enableGlobalFilter: true,
+    state: {
+      globalFilter,
+    },
+    initialState: {
+      rowSelection,
+      pagination: {
+        pageSize: 25,
+      },
+    },
+  });
+
+  const addVendors = async () => {
+    const vendorIds = tableInstance
+      .getSelectedRowModel()
+      .rows.filter((r) => !r.original.linked_system)
+      .map((r) => r.original.vendor_id);
+    if (vendorIds.length > 0) {
+      await postVendorIds(vendorIds);
+      router.push(redirectRoute);
+      toast(
+        successToastParams(
+          `Successfully added ${
+            vendorIds.length
+          } ${systemText.toLocaleLowerCase()}`
+        )
+      );
+    }
+  };
+
+  const anyNewSelectedRows = tableInstance
+    .getSelectedRowModel()
+    .rows.some((row) => !row.original.linked_system);
+
+  if (isPostLoading || isPostSuccess) {
+    return (
+      <Flex height="100%" justifyContent="center" alignItems="center">
+        <Spinner />
+      </Flex>
+    );
+  }
+
+  if (isGetLoading) {
+    return <TableSkeletonLoader rowHeight={36} numRows={15} />;
+  }
+
+  const toolTipText = allRowsAdded
+    ? `All ${systemText.toLocaleLowerCase()} have already been added`
+    : `Select a ${systemText.toLocaleLowerCase()} `;
+
+  return (
+    <Flex flex={1} direction="column" overflow="auto">
+      <ConfirmationModal
+        isOpen={isOpen}
+        isCentered
+        onCancel={onClose}
+        onClose={onClose}
+        onConfirm={addVendors}
+        title="Confirmation"
+        message={`You are about to add ${
+          tableInstance
+            .getSelectedRowModel()
+            .rows.filter((r) => !r.original.linked_system).length
+        } ${systemText.toLocaleLowerCase()}`}
+      />
+      <TableActionBar>
+        <GlobalFilterV2
+          globalFilter={globalFilter}
+          setGlobalFilter={setGlobalFilter}
+          placeholder="search"
+        />
+        <Tooltip
+          label={toolTipText}
+          shouldWrapChildren
+          placement="top"
+          isDisabled={
+            (anyNewSelectedRows && !allRowsAdded) ||
+            (anyNewSelectedRows && allRowsAdded)
+          }
+        >
+          <Button
+            onClick={onOpen}
+            size="xs"
+            variant="outline"
+            disabled={!anyNewSelectedRows}
+          >
+            Add {`${systemText}s`}
+          </Button>
+        </Tooltip>
+      </TableActionBar>
+      <FidesTableV2<MultipleSystemTable>
```

Comment:
another side note: The typing is awesome in version 8 of the table library!! The author definitely put a lot of effort into making the typing experience nice. I'm pretty sure everything is strongly typed.

You get autocomplete for the row fields.